dor/providers/parsers.py

```diff
@@ -11,28 +14,31 @@ class DescriptorFileParser:
         "PREMIS": "http://www.loc.gov/premis/v3",
     }
 
-    def __init__(self, descriptor_path):
-        text = descriptor_path.read_text()
+    def __init__(self, descriptor_file_path: Path, file_provider: FileProvider):
+        text = descriptor_file_path.read_text()
         self.tree: ElementAdapter = ElementAdapter.from_string(text, self.namespaces)
-
+        self.file_provider = file_provider
+        self.data_path = file_provider.get_data_dir(descriptor_file_path)
+        
     def get_id(self):
         return uuid.UUID(self.tree.get("OBJID"))
 
     def get_type(self):
         hdr = self.tree.find("METS:metsHdr")
         return hdr.get("TYPE")
 
-    def get_alternate_identifier(self):
+    def get_alternate_identifier(self)-> Union[AlternateIdentifier, None]:
         alt_record_id = self.tree.find("METS:metsHdr/METS:altRecordID")
+        
         if alt_record_id:
             return AlternateIdentifier(
                 type=alt_record_id.get("TYPE"), id=alt_record_id.text
-            )
+            ) 
```

Comment:
I mentioned something before I think, but `self.tree.find` will raise an exception if it doesn't find what it's looking for. So the `if alt_record_id` check is unnecessary, and you can just return an `AlternateIdentifier`